pyston/utils/files.py

```diff
@@ -24,7 +32,34 @@ class InvalidResponseStatusCode(SuspiciousOperation):
     pass
 
 
-def get_file_content_from_url(url, limit, timeout=1):
+def get_content_type_from_filename(filename):
+    return mimetypes.guess_type(filename)[0]
+
+
+def get_content_type_from_file_content(content):
+    with magic.Magic(flags=magic.MAGIC_MIME_TYPE) as m:
+        mime_type = m.id_buffer(content.read(1024))
+        content.seek(0)
+        return mime_type
+
+
+def get_filename_from_content_type(content_type):
+    extensions = mimetypes.guess_all_extensions(content_type)
+    known_extensions = [ext for exts, name in settings.DEFAULT_FILENAMES for ext in exts]
+    extensions.sort(
+        key=lambda x: known_extensions.index(x[1:]) if x[1:] in known_extensions else len(known_extensions)
+    )
```

Comment:
tohle je tu důležité jelikož mimetypes.guess_all_extensions nedává vždy stejné pořadí, proto dále vylepšujeme pořadí podle pořadí definované v DEFAULT_FILENAMES